testnet/stacks-node/src/nakamoto_node/relayer.rs

```diff
@@ -1032,7 +995,12 @@ impl RelayerThread {
         debug!("Relayer: successfully stopped tenure.");
 
         // Get the necessary snapshots and state
-        let burn_tip = self.get_block_snapshot_consensus(&new_burn_view)?;
+        let burn_tip =
+            SortitionDB::get_block_snapshot_consensus(self.sortdb.conn(), &new_burn_view)?
+                .ok_or_else(|| {
+                    error!("Relayer: failed to get block snapshot for new burn view");
+                    NakamotoNodeError::SnapshotNotFoundForChainTip
+                })?;
```

Comment:
In this case, I removed the wrapper function, and I also removed the `map_err`. The `map_err` is only for `DBError`, which IMO deserves a different `NakamotoNodeError` (which I've added) and (again IMO) doesn't need a custom error log.

Reply:
(Same exact story with a few lines below)